wpimath/src/main/native/cpp/geometry/Quaternion.cpp

```diff
@@ -69,20 +71,24 @@ Quaternion Quaternion::operator*(const Quaternion& other) const {
 }
 
 bool Quaternion::operator==(const Quaternion& other) const {
-  return std::abs(W() * other.W() + m_v.dot(other.m_v)) > 1.0 - 1E-9;
+  return std::abs(std::abs(Dot(other)) - Norm() * other.Norm()) < 1e-9;
```

Comment:
Is this change missing in Java?

Reply:
I actually meant for the change in java to make it here - I think it makes more sense for the general quaternion class to do a strict equality while the Rotation3d class checks for equality in SO(3).